0xPolygon/polygon-edge
# [ Feature ] Add support for GCP secrets manager

# Description

This PR adds the support for storing secrets in Google Cloud Platform secrets manager.    

A new `--type` option in the `secrets generate` command has been added - `gcp-ssm`.
   
When running `secrets generate` command the operator needs to specify two new parameters in `--extra` flag - `project-id` - the project id provided in GCP dashboard and `gcp-ssm-cred` - json file with credentials exported from GCP console


# Changes include


# Checklist


## Testing


### Manual tests

In order to test this, first of all you'll need to go to [GCP Console](https://console.cloud.google.com) and enable Billing Account for your Gmail account but  there is also free trial available if you're using GCP for the first time. 

Next, find and enable [Secrets Manager API](https://console.cloud.google.com/apis/library/browse) and get the credentials from GCP console by following these [instructions ](https://cloud.google.com/secret-manager/docs/reference/libraries)

Generate `secretsManagerConfig.json` file by running `polygon-edge secrets generate --name node1 --extra project-id=<your_project_id>,gcp-ssm-cred=<credentials_json_file> --type gcp-ssm`   
Store secrets in GCP SM by running `secrets init --config secretsManagerConfig.json`
You should now see the stored secrets in GCP Secrets Manager console.

When starting the server, among standard flags, we need to specify `--secrets-config secretsManagerConfig.json`


# Documentation update

Public documentation PR link: https://github.com/0xPolygon/polygon-edge-docs/pull/86


## Additional Comments
Fixes EDGE-508
